asf_public_discourse_home_decarbonisation/pipeline/data_processing_flows/flow_utils.py

```diff
@@ -0,0 +1,267 @@
+
+    Args:
+        category (str): An MSE category, "sample" or "all"
+        collection_datetime (str): A date in the format "YYYY_MM_DD"
+
+    Returns:
+        pd.DataDrame: a dataframe with the MSE data
+    """
+    mse_categories = [
+        "green-ethical-moneysaving",
+        "lpg-heating-oil-solid-other-fuels",
+        "energy",
+        "is-this-quote-fair",
+    ]
+
+    accepted_categories = mse_categories + ["all"]
+    try:
+        # Check if the category is in the list if one of the accepted categories
+        category_index = accepted_categories.index(category)
+        if category in [
+            "green-ethical-moneysaving",
+            "lpg-heating-oil-solid-other-fuels",
+            "is-this-quote-fair",
+            "energy",
+        ]:
+            mse_data = get_mse_category_data(category, collection_datetime)
+        else:  # category == "all"
+            mse_data = get_all_mse_data(collection_datetime)
+        logger.info(f"Data for category '{category}' imported successfully from S3.")
+    except ValueError:
+        logger.error(f"{category} is not a valid category!")
+        sys.exit(-1)
+
+    return mse_data
+
+
+def remove_urls(text: str) -> str:
+    """
+    Removes URLs from text.
+    Args:
+        text (str): a string, tipically one or multiple sentences long
+    Returns:
+        str: text without URLs
+    """
+    pattern = r"http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\\(\\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+"
+    cleaned_text = re.sub(pattern, " ", text)
+    return cleaned_text
+
+
+def remove_text_after_patterns(text: str) -> str:
+    """
+    Removes pattern of the form "xxx writes: ".
+
+    Args:
+        text (str): text to be cleaned
+
+    Returns:
+        str: cleaned text
+    """
+    # We use re.sub() to replace the pattern with an empty string
+    result = re.sub(r"\w+ wrote »", " ", text)
+    return result
+
+
+def remove_items_in_list(mse_data: pd.DataFrame, list_of_items: list) -> pd.DataFrame:
+    """
+    Removes items from text and titles.
+
+    Args:
+        mse_data (pd.DataFrame): Money Saving Expert data
+        list_of_items (list): list of items to be removed
+    Returns:
+        pd.DataFrame: MSE data with items removed from text and titles
+    """
+    mse_data["tokens_title"] = mse_data["tokens_title"].apply(
+        lambda x: [token for token in x if token not in list_of_items]
+    )
+    mse_data["tokens_text"] = mse_data["tokens_text"].apply(
+        lambda x: [token for token in x if token not in list_of_items]
+    )
+    return mse_data
+
+
+def english_stopwords_definition() -> list:
+    """
+    Defines English stopwords by putting together NLTK and gensim stopwords.
+
+    Returns:
+        list: a list of English stopwords.
+    """
+    sw_nltk = stopwords.words("english")
+    sw_gensim = [s for s in STOPWORDS if s not in sw_nltk]
+
+    stopwords_list = sw_nltk + sw_gensim
```

Comment:
Wondering if something like this operation would be quicker than doing a for loop?
"first_list = ["apple", "banana", "cherry"]
second_list = ["banana", "dragonfruit", "elderberry"]

resultList = list(set(first_list) | set(second_list))" and the resulting output is: "['banana', 'dragonfruit', 'cherry', 'elderberry', 'apple']"

Reply:
Can do - but these lists are so small, it doesn't really make a difference.